examples/self-diagnostics/src/main.rs

```diff
@@ -13,6 +13,9 @@ fn init_meter_provider() -> opentelemetry_sdk::metrics::SdkMeterProvider {
     let reader = PeriodicReader::builder(exporter, opentelemetry_sdk::runtime::Tokio).build();
 
     let provider = opentelemetry_sdk::metrics::SdkMeterProvider::builder()
+        .with_resource(opentelemetry_sdk::resource::Resource::new(vec![
```

Comment:
There is no need to allocate a `vec` here. Since this is an example, it's good to showcase good practices.

```suggestion
        .with_resource(opentelemetry_sdk::resource::Resource::new([
```